internal/cmd/kubernetes.go

```diff
@@ -151,12 +151,23 @@ func writeKubeconfig(user *api.User, destinations []api.Destination, grants []ap
 			CertificateAuthorityData: ca,
 		}
 
-		kubeConfig.Contexts[context] = &clientcmdapi.Context{
-			Cluster:   context,
-			AuthInfo:  user.Name,
-			Namespace: namespace,
+		kubeContext, ok := kubeConfig.Contexts[context]
```

Comment:
nit: Could be nice to have a comment for future readers so that they understand this is preserving whatever namespaces the user has defined in their default context.